File: _insert_interval.ts
function insert(intervals: number[][], nI: number[]): number[][] 
{
    
    let left = []
    let right = []
    let newTemp=[nI[0],nI[1]]
    
    for(let cur of intervals){
        if(cur[1]< nI[0]){
            left.push(cur)
        }
        else if(cur[0]>nI[1]){
            right.push(cur)
        }
        else{
            let low = Math.min(
                cur[0],nI[0]
            )
            let high = Math.max(
                cur[1],nI[1]
            )
            newTemp[0] = low
            newTemp[1] = high
            nI = [low,high]
        }
    }
    
    // if(intervals.length==0)
    //     return [nI]

    return left.concat([newTemp]).concat(right)
};
          
//good ref -https://leetcode.com/problems/insert-interval/solutions/844394/python-js-go-c-o-n-by-boundary-checking-w-comment/?orderBy=most_votes&languageTags=javascript

File: Python/insert_interval.py
class Solution:
    def insert(self, intervals: List[List[int]], nI: List[int]) -> List[List[int]]:

        left = []
        right = []
        mid = nI

        for i in range(0,len(intervals)):
            cur = intervals[i]

            if(cur[1]<nI[0]):
                left.append(cur)
            elif(cur[0]>nI[1]):
                right.append(cur)
            else:
                first = min(cur[0], mid[0])
                second = max(cur[1], mid[1])

                mid = [first, second]
        
        left.append(mid)
        
        return left + right
